File: zmisc/443_string_compression.py
class Solution:
    def compress(self, chars: list[str]) -> int:
        length = len(chars)
        count = 1
        lst = []
        after = ''
        prev = ''
        for i in range(length):
            if i == length - 1:
                after  = ''
            else:
                after = chars[i + 1]

            if chars[i] == after:
                count += 1
            else:
                if count == 1: 
                    lst.append(str(chars[i]))
                else:
                    lst.append(prev)
                    if count < 10:
                        lst.append(str(count))
                    else:
                        llst = []
                        while count:
                            remainder = count % 10
                            count = count // 10
                            llst.append(str(remainder))
                        for j in reversed(llst):
                            lst.append(j)
                    count = 1
            prev = chars[i]
        print(lst)


chars = ["a","a","b","b","c","c","c"]
print(Solution().compress(chars))

File: array/443_string_compression.java
class Solution {
    public int compress(char[] chars) {
        int res = 0, i = 0;
        while (i < chars.length) {
            int count = 1;
            while (i + count < chars.length && chars[i + count] == chars[i]) {
                count++;
            }
            chars[res++] = chars[i];
            if (count > 1) {
                for (char c : Integer.toString(count).toCharArray()) {
                    chars[res++] = c;
                }
            }
            i += count;
        }
        return res;
    }
}